tests/unit_tests/test_time_series_any_time_series.py

```diff
@@ -0,0 +1,293 @@
+    # act
+    result = any_time_series.index_for_timestamp_not_earlier_than('2016-02-02T00:00:00.588Z')
+    # assert
+    assert result is None
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2015-02-02T00:00:00.588Z']])
+def test_index_for_timestamp_not_earlier_than_same_timestamp(timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    # act
+    result = any_time_series.index_for_timestamp_not_earlier_than('2015-02-02T00:00:00.588Z')
+    # assert
+    assert result == 1
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2015-02-02T00:00:00.588Z']])
+def test_index_for_timestamp_closest_to_empty_timestamps(timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = None
+    # act
+    result = any_time_series.index_for_timestamp_closest_to('2000-01-01T00:00:00.588Z')
+    # assert
+    assert result is None
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2015-02-02T00:00:00.588Z']])
+def test_index_for_timestamp_closest_to_timestamp_too_early(timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    # act
+    result = any_time_series.index_for_timestamp_closest_to('2013-02-02T00:00:00.588Z')
+    # assert
+    assert result == 0
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2015-02-02T00:00:00.588Z']])
+def test_index_for_timestamp_closest_to_timestamp_len(mocker, timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    index_for_timestamp_not_later_mock = mocker.Mock(return_value = 1)
+    any_time_series.index_for_timestamp_not_later_than = index_for_timestamp_not_later_mock
+    # act
+    result = any_time_series.index_for_timestamp_closest_to('2013-02-02T00:00:00.588Z')
+    # assert
+    assert result == 1
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2015-02-02T00:00:00.588Z']])
+def test_index_for_timestamp_closest_to_timestamp_equals_before(mocker, timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    index_for_timestamp_not_later_mock = mocker.Mock(return_value = 0)
+    any_time_series.index_for_timestamp_not_later_than = index_for_timestamp_not_later_mock
+    # act
+    result = any_time_series.index_for_timestamp_closest_to('2014-01-01T00:00:00.588Z')
+    # assert
+    assert result == 0
+    index_for_timestamp_not_later_mock.assert_called_once_with('2014-01-01T00:00:00.588Z')
```

Comment:
I disagree with the use of mocker functions here.  Simply test the functionality of the index_for_timestamp_closest_to() method without checking how many times it has called other internal methods.